docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,92 @@
+# Allowing Groups to be Based on and Synced With External LDAP Records
+
+Linking internal `Group`s of users to their representations on LDAP servers allows for administrators to have an easier way to manage groups inside OpenShift. A command like `oadm sync groups` or `oadm sync group <name>` will trigger a sync of group members with their external support (if any).
+
+## Supporting Group Sources
+In order to link `Group` structures to external providers of their data, each `Group` will need an (optional) `GroupSource`:
+
+```go
+type Group struct{
+  ...
+  // Source is an optional parameter that enables the Group member list to be linked to an entry in another system or database
+  Source GroupSource
+}
+
+// GroupSource enables the Group member list to be linked to an entry in another system or database
+type GroupSource interface{
+  parent *Group
+  // Sync updates the group's list of members to reflect the parallel list in the remote system or database
+  Sync() (deleted bool, members []string, err error)
```

Comment:
I think this should end up being client logic.  The syncing program will look at the OpenShiftGroup, get the list of OpenShiftUsers in that group, the LDAPGroup, the list of LDAPUsers in the LDAPGroup, find the OpenShiftIdentities for the LDAPUsers, find the OpenShiftUsers for the OpenShiftIdentities, determine the new set of OpenShiftUsers for the OpenShiftGroup and stomp the contents.